### PR TITLE
Make attached surfaces respect exclusive zones

### PR DESCRIPTION
**Pull request by wmww**
Before they were respecting exclusive zones when they were attached to one side, but when they were attached to opposite sides they anchored to the screen edges instead. This makes the behavior consistent and fixes an upcoming WLCS test.

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
If you change a test called `stretched_attached_window_ignores_exclusive_zone` to test that exclusive zones are respected, then it is probably good to rename the test.

File: tests/miral/window_placement_attached.cpp [hunk 1]
```diff
@@ -426,7 +426,7 @@ TEST_P(WindowPlacementAttached, window_ignores_exclusive_zone_when_fullscreened)
     EXPECT_THAT(normal.size(), Eq(display_area.size));
 }
 
-TEST_P(WindowPlacementAttached, stretched_attached_window_ignores_exclusive_zone)
+TEST_P(WindowPlacementAttached, stretched_attached_window_respects_exclusive_zone)
```
Comment (wmww) [inline tests/miral/window_placement_attached.cpp:429]: I... did rename the test?

**Review by AlanGriffiths — APPROVED** [review]
bors r+

File: tests/miral/window_placement_attached.cpp (hunk 1 above)
Comment (AlanGriffiths) [reply tests/miral/window_placement_attached.cpp:429]: Weird. I see it now.

:shrug:

**Pull request merged**